### ISSUE TITLE
一个很关键的问题：在进行CSP的时候，测试集数据的特征是否被混入训练特征计算CSP

### ISSUE THREAD

**Issue by sjx66606** [fchest/DARNet#3]
在进行CSP提取特征的时候并没有区分训练集和测试集，而是先对所有数据提取CSP特征，然后又区分训练集特征和测试集特征。这样的话，测试集的信息是否在训练集的特征中有所包含？

**Comment by FHASLKJFHEWQOI** [comment]
Yes, this method presents a risk of information leakage.  The current code contains issues that may lead to inaccurate or overly optimistic results, and it is crucial to address these problems to ensure the reported outcomes are fair and reliable.

The authors should revise all the results in the current version before submitting the final version to NeurIPS.

**Comment by fchest** [comment]
This CSP processing method is consistent with that of DBPNet. Upon review, we confirmed the presence of data leakage. In the final version submitted to NeurIPS, we have corrected the relevant results, and the updated code is now available. We appreciate you bringing these issues to our attention.
In the supplement experimental,  we rigorously avoided risk of data leakage. Experimental results demonstrate that our model consistently achieves SOTA performance across most test conditions.
![Correction](https://github.com/user-attachments/assets/37125249-627d-4107-a027-12072e8aeb38)
